robolectric/src/test/java/org/robolectric/shadows/ShadowPaintTest.java

```diff
@@ -78,6 +78,27 @@ public void measureTextActuallyMeasuresLength() {
     assertThat(paint.measureText(new StringBuilder("Hello"), 1, 4)).isEqualTo(3.0f);
   }
 
+  @Test
+  public void measureTextUsesTextScaleX() {
+    Paint paint = new Paint();
+    paint.setTextScaleX(1.5f);
+    assertThat(paint.measureText("Hello")).isEqualTo(7.5f);
+    assertThat(paint.measureText("Hello", 1, 3)).isEqualTo(3.0f);
+    assertThat(paint.measureText(new StringBuilder("Hello"), 1, 4)).isEqualTo(4.5f);
+  }
+
+  @Test
+  public void textWidthWithNegativeScaleXIsTheSameAsPositive() {
+    Paint positive = new Paint();
+    Paint negative = new Paint();
+    positive.setTextScaleX(1.5f);
+    negative.setTextScaleX(-1.5f);
+    assertThat(positive.measureText("Hello")).isEqualTo(negative.measureText("Hello"));
```

Comment:
Maybe we can add a new assertion to test final `measureText` result, e.g.

```
assertThat(positive.measureText("Hello")).isEqualsTo(7.5f);
```

Reply:
It will duplicates another test in this class, isn't?